mainwindow.cc

```diff
@@ -3723,6 +3764,14 @@ void MainWindow::unzoom()
   applyZoomFactor();
 }
 
+void MainWindow::viewLinkClicked( const QUrl & url )
+{
+  if( scanPopup.get() && scanPopup->isVisible() )
```

Comment:
What if pinned scan popup is visible but not the active window, and the user clicks a link in the main window?

Reply:
And please don't just add a check if the scan popup window or the main window is active. That would also be a hack/workaround, not a proper solution. Is a single article view shared by the main window and the scan popup now? Or is only the request interceptor shared? Either use different objects or reliably disambiguate the intercepted requests, e.g. by comparing pointers to article views or object names.